bridge/client/app/_views/ktb-settings-view/ktb-integration-view/ktb-integration-view.component.spec.ts

```diff
@@ -40,4 +61,275 @@ describe('KtbIntegrationViewComponent', () => {
+      })
+    );
+
+    // when
+    const actual = await firstValueFrom(component.params$);
+
+    // then
+    expect(actual.projectName).toBe('sockshop');
+    expect(actual.integrationId).toBe('abcdef');
+  });
+
+  it('should emit the selected uniform registration when the path param is provided', async () => {
+    // given
+    paramsSubject.next(
+      convertToParamMap({
+        projectName,
+        integrationId: 'abcdef',
+      })
+    );
+
+    // when
+    const actual = await firstValueFrom(component.selectedUniformRegistrationId$);
+
+    // then
+    expect(actual).toEqual({ id: 'abcdef' });
+  });
+
+  it('should load uniform registrations and build the data source', async () => {
+    // given
+    // when
+    const dataSource = await firstValueFrom(component.uniformRegistrations$);
+
+    // then
+    expect(dataSource.data).toBeTruthy();
+    expect(dataSource.data.length).toBe(11);
+  });
+
+  it('should select a uniform registration by the user', (done) => {
+    // given
+    const id = 'af567de129';
+    const u = new UniformRegistration();
+    u.id = id;
+    component.selectedUniformRegistrationId$.subscribe((actual) => {
+      // then
+      expect(actual).toEqual({ id });
+      done();
+    });
+
+    // when
+    component.setSelectedUniformRegistration(u, projectName);
+  });
+
+  it('should pre select a uniform registration when the path param is set', async () => {
+    // given
+    const integrationId = 'keptn-lighthouse-service-8feec7146c19fa08bd65664b8d47f153';
+    paramsSubject.next(
+      convertToParamMap({
+        projectName,
+        integrationId,
+      })
+    );
+
+    // when
+    const actual = await firstValueFrom(component.selectedUniformRegistration$);
+
+    // then
+    expect(actual).toBeTruthy();
+    expect(actual?.id).toBe(integrationId);
+  });
+
+  it('should pre load logs', async () => {
+    // given
+    const integrationId = 'keptn-lighthouse-service-8feec7146c19fa08bd65664b8d47f153';
+    paramsSubject.next(
+      convertToParamMap({
+        projectName,
+        integrationId,
+      })
+    );
+
+    // when
+    const actual = await firstValueFrom(component.uniformRegistrationLogs$);
+
+    // then
+    expect(actual).toBeTruthy();
+    expect(actual.length).toBe(10);
+  });
+
+  it('should get subscriptions for a uniform registration', () => {
+    // given
+    const u = new UniformRegistration();
+    const s1 = new UniformSubscription();
+    s1.event = 'Dog';
+    const s2 = new UniformSubscription();
+    s2.event = 'Cat';
+    u.subscriptions = [s1, s2];
+
+    // when
+    const actual = component.getSubscriptions(u, projectName);
+
+    // then
+    expect(actual).toEqual(['Dog', 'Cat']);
+  });
+
+  it('should convert unknown to a uniform registration', () => {
+    // given
+    const u: unknown = new UniformRegistration();
+
+    // when
+    const actual = component.toUniformRegistration(u);
+
+    // then
+    expect(actual).toBeTruthy();
+  });
```

Comment:
Such type conversions are not really testable, but I would suggest maybe comparing the input and the output so that it is the same object and the same reference.
```suggestion
expect(actual).toBe(u);
expect(actual).toEqual(new UniformRegistration());
```

Reply:
Changed